sdk/core/Azure.Core/tests/common/ModelReaderWriter/Models/DiscriminatorSet/BaseModel.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net.ClientModel;
+using System.Net.ClientModel.Core;
+using System.Text.Json;
+using Azure.Core.Tests.Common;
+
+namespace Azure.Core.Tests.ModelReaderWriterTests.Models
+{
+    [ModelReaderProxy(typeof(UnknownBaseModel))]
+    public abstract class BaseModel : IUtf8JsonSerializable, IJsonModel<BaseModel>
+    {
+        private Dictionary<string, BinaryData> _rawData;
+
+        protected internal BaseModel(Dictionary<string, BinaryData> rawData)
+        {
+            _rawData = rawData ?? new Dictionary<string, BinaryData>();
+        }
+
+        public string Kind { get; internal set; }
+        public string Name { get; set; }
+
+        protected internal void SerializeRawData(Utf8JsonWriter writer)
+        {
+            //write out the raw data
+            foreach (var property in _rawData)
+            {
+                writer.WritePropertyName(property.Key);
+#if NET6_0_OR_GREATER
```

Comment:
We should add a helper to (System.ClientModel) that does the runtime switching so that model implementations don't have to multitarget.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/39506
https://github.com/Azure/autorest.csharp/issues/3867

I think we can do this in the `feature/core-experiment` branch.

Reply:
@KrzysztofCwalina are you happy shipping without this comment addressed in the GA package?

Reply:
yes